src/main/java/wolox/training/services/BookService.java

```diff
@@ -27,32 +27,48 @@ public BookService(BookRepository bookRepository, OpenLibraryDelegate openLibrar
 
     }
 
-    public Iterable<Book> findAll(String isbn) {
+    public Iterable<Book> findAll(Map<String, String> params) {
 
-        if (StringUtils.isNotEmpty(isbn)) {
+        if (!params.isEmpty()) {
 
-            return filterBookByIsbn(isbn);
+            if (params.get("isbn") != null) {
```

Comment:
I think you can use the `containsKey` method instead of checking this null, but it's ok.